src/dev/impl/DevToys/ViewModels/Tools/Generators/CheckSumGenerator/CheckSumGeneratorToolViewModel.cs

```diff
@@ -0,0 +1,432 @@
+        {
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                DataPackageView? dataPackageView = Clipboard.GetContent();
+
+                IStorageItem? storageItem = await ExtractStorageItem(dataPackageView);
+                if (storageItem is StorageFile file)
+                {
+                    InputFile = file;
+                }
+            }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region CancelCommand
+
+        public IRelayCommand CancelCommand { get; }
+
+        private void ExecuteCancelCommand(bool shouldRevertHashingAlgo)
+        {
+            lock (_lockObject)
+            {
+                _hasCancelledCalculation = true;
+                _cancellationTokenSource.Cancel();
+                _cancellationTokenSource.Dispose();
+
+                _cancellationTokenSource = new CancellationTokenSource();
+
+                if (shouldRevertHashingAlgo && _outputHashingAlgorithm != null)
+                {
+                    InputHashingAlgorithm = _outputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                _hasCancelledCalculation = false;
+                ShouldDisplayProgress = false;
+                ResetProgress();
+            }
+        }
+
+        #endregion
+
+        private void UpdateOutputCase()
+        {
+            if (string.IsNullOrWhiteSpace(Output))
+            {
+                return;
+            }
+
+            if (IsUppercase)
+            {
+                Output = Output!.ToUpperInvariant();
+            }
+            else
+            {
+                Output = Output!.ToLowerInvariant();
+            }
+        }
+
+        private void CheckSum()
+        {
+            lock (_lockObject)
+            {
+                ExecuteCancelCommand(shouldRevertHashingAlgo: false);
+                CheckSumAsync().Forget();
+            }
+        }
+
+        private async Task CheckSumAsync()
+        {
+            if (_isCalculationInProgress || InputFile is null)
+            {
+                return;
+            }
+
+            _isCalculationInProgress = true;
+
+            await TaskScheduler.Default;
+
+            using Stream? fileStream = await InputFile.OpenStreamForReadAsync();
+
+            await ThreadHelper.RunOnUIThreadAsync(() => ShouldDisplayProgress = HashingHelper.ComputeHashIterations(fileStream) > _displayAboveIterations);
+
+            string? hashOutput = await CalculateFileHash(InputHashingAlgorithm.Value, fileStream);
+
+            await ThreadHelper.RunOnUIThreadAsync(() =>
+            {
+                if (hashOutput != null)
+                {
+                    Output = hashOutput;
+                    _outputHashingAlgorithm = InputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                ShouldDisplayProgress = false;
```

Comment:
I'm facing a race condition on my machine and I think that this line along with 343 and maybe a few others are the cause.

The scenario I have is:
1. Select a file. Computation start. Progress bar is displayed.
2. Select another file while the computation of the 1st one isn't done yet.
3. In this scenario, the progress bar simply disappear.

I think the race here is that line 343 (with the new file) runs before 356 (with the old file) is executed.

A proper fix is probably to cache the task on line 326 and await it (when not null) right after line 339. Thoughts?